ngraph/core/src/op/dft.cpp

```diff
@@ -0,0 +1,259 @@
+        this, num_of_inputs == 2 || num_of_inputs == 3, "DFT must have 2 or 3 inputs.");
+
+    element::Type input_et = get_input_element_type(0);
+    NODE_VALIDATION_CHECK(this,
+                          input_et == element::f32 || input_et == element::f16 ||
+                              input_et == element::bf16,
+                          "DFT input element type must be f32, f16, or bf16");
+
+    element::Type axes_et = get_input_element_type(1);
+    NODE_VALIDATION_CHECK(this,
+                          axes_et == element::i64 || axes_et == element::i32,
+                          "DFT axes element type must be i32 or i64");
+
+    PartialShape input_shape = PartialShape(get_input_partial_shape(0));
+    if (input_shape.rank().is_static())
+    {
+        const auto input_rank = input_shape.rank().get_length();
+        NODE_VALIDATION_CHECK(this,
+                              input_rank >= 2,
+                              "The input rank must be greater or equal to 2. Got input rank: ",
+                              input_rank);
+
+        auto last_dim_with_two = input_shape[input_rank - 1] & Dimension(2);
+        NODE_VALIDATION_CHECK(this,
+                              !last_dim_with_two.get_interval().empty(),
+                              "The last dimension of input data must be 2. Got: ",
+                              input_shape[input_rank - 1]);
+    }
+
+    PartialShape axes_shape = PartialShape(get_input_partial_shape(1));
+    if (axes_shape.rank().is_static())
+    {
+        NODE_VALIDATION_CHECK(this,
+                              axes_shape.rank().get_length() == 1,
+                              "DFT axes input must be 1D tensor. Got axes input rank: ",
+                              axes_shape.rank().get_length());
+    }
+
+    if (input_shape.rank().is_static() && axes_shape.is_static())
+    {
+        const auto input_rank = input_shape.rank().get_length();
+        NODE_VALIDATION_CHECK(this,
+                              input_rank >= axes_shape.to_shape()[0] + 1,
+                              "The input rank must be greater than number of DFT axes. Got "
+                              "input rank: ",
+                              input_rank,
+                              ", number of axes: ",
+                              axes_shape.to_shape()[0]);
+    }
+
+    if (input_shape.rank().is_static() && is_type<op::Constant>(input_value(1).get_node()))
+    {
+        const auto input_rank = input_shape.rank().get_length();
+        const auto& const_axes = get_constant_from_source(input_value(1));
+        auto axes = const_axes->cast_vector<int64_t>();
+
+        for (int64_t& axis : axes)
+        {
+            if (axis < 0)
+            {
+                axis += input_rank - 1;
+            }
+        }
+
+        AxisVector axes_vector;
+        AxisSet axes_set;
+        for (const int64_t axis : axes)
+        {
+            axes_vector.push_back(static_cast<size_t>(axis));
```

Comment:
Why do you need `axes_vector`? You already have a vector `axes` so you can use it to check that all elements are unique.

Reply:
The `axes_vector` is used to print diagnostic in the `NODE_VALIDATION_CHECK` at lines 147--148, because there are no `std::ostream& operator(std::ostream& stream, const std::vector<int64_t>& v)`, but there is `std::ostream& operator(std::ostream& stream, const AxisVector& v)`.